src/main/java/omero/gateway/facility/RawDataFacility.java

```diff
@@ -201,17 +207,78 @@ public Plane2D getPlane(SecurityContext ctx, PixelsData pixels, int z,
     public Plane2D getTile(SecurityContext ctx, PixelsData pixels, int z,
             int t, int c, int x, int y, int w, int h)
             throws DataSourceException {
+        return getTile(ctx, pixels, z, t, c, x, y, w, h, -1);
+    }
+
+    /**
+     * Extracts a 2D tile from the pixels set
+     *
+     * @param ctx
+     *            The security context.
+     * @param pixels
+     *            The {@link PixelsData} object to fetch the data from.
+     * @param z
+     *            The z-section at which data is to be fetched.
+     * @param t
+     *            The timepoint at which data is to be fetched.
+     * @param c
+     *            The channel at which data is to be fetched.
+     * @param x
+     *            The x coordinate
+     * @param y
+     *            The y coordinate
+     * @param w
+     *            The width of the tile
+     * @param h
+     *            The height of the tile
+     * @param resolutionLevel
+     *            A specific resolution level (optional), see also
+     *            getResolutionDescriptions()
+     * @return A plane 2D object that encapsulates the actual tile pixels.
+     * @throws DataSourceException
+     *             If an error occurs while retrieving the plane data from the
+     *             pixels source.
+     */
+    public Plane2D getTile(SecurityContext ctx, PixelsData pixels, int z,
+                           int t, int c, int x, int y, int w, int h, int resolutionLevel)
+            throws DataSourceException {
         if (pixels == null)
             return null;
-        
+
         try {
-            return getDataSink(ctx, pixels, gateway).getTile(z, t, c, x, y, w,
-                    h);
+            DataSink ds = getDataSink(ctx, pixels, gateway);
+            int oldResLevel = -1;
+            if (resolutionLevel >= 0)
+                oldResLevel = ds.setResolutionLevel(resolutionLevel);
+            Plane2D tile = ds.getTile(z, t, c, x, y, w, h);
+            if (resolutionLevel >= 0)
+                ds.setResolutionLevel(oldResLevel);
+            return tile;
         } catch (DSOutOfServiceException e) {
             throw new DataSourceException("Can't initiate DataSink", e);
         }
     }
 
+    /**
+     * Get the available resolution descriptions
+     * @param ctx The SecurityContext
+     * @param pixels The PixelsData object
+     * @return See above
+     * @throws DataSourceException If an error occurs while retrieving the data
+     */
+    public List<ResolutionDescription> getResolutionDescriptions(SecurityContext ctx, PixelsData pixels)
+            throws DataSourceException {
+        if (pixels == null)
+            return null;
+
+        try {
+            return getDataSink(ctx, pixels, gateway).getResolutionDescriptions();
```

Comment:
maybe a method initialising the ``datasink`` could be created so the error below could be sent.

Reply:
I'm not sure what you mean. The `getDataSink` method gets a 'dataset' from the cache or creates a new one if it doesn't exist.

Reply:
The error thrown is for that method ``cannot init dataSink``
it will be good to do the same for the other methods